From 36449131351e96f8c7ac4e6e5f8c746bb6437be6 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 13 Dec 2023 17:04:49 +0800
Subject: [PATCH 202/232] misc: rk628: post_process: Fix scaler timing
 calcalculation error

Assume that Fout is the output pixel clock frequency
and Fin is the input pixel clock frequency:

dsp_in_vtotal * dsp_in_htotal * (1 / Fin) = dsp_out_vtotal * dsp_out_htotal * (1 / Fout)

Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Change-Id: I77055cdf47f49de5ae7b5eb94b257e07171dd347
Signed-off-by: Guochun Huang <hero.huang@rock-chips.com>
---
 drivers/misc/rk628/rk628_post_process.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/rk628/rk628_post_process.c b/drivers/misc/rk628/rk628_post_process.c
index a27f43698bad3d..66f9d63e6056ed 100644
--- a/drivers/misc/rk628/rk628_post_process.c
+++ b/drivers/misc/rk628/rk628_post_process.c
@@ -268,8 +268,8 @@ void rk628_post_process_init(struct rk628 *rk628)
 	u64 dst_rate, src_rate;
 
 	src_rate = src->clock * 1000;
-	dst_rate = src_rate * dst->vdisplay * dst->htotal;
-	do_div(dst_rate, (src->vdisplay * src->htotal));
+	dst_rate = src_rate * dst->vtotal * dst->htotal;
+	do_div(dst_rate, (src->vtotal * src->htotal));
 	do_div(dst_rate, 1000);
 	printk("src %dx%d clock:%d\n",
 	       src->hdisplay, src->vdisplay, src->clock);
-- 
2.17.1

